msal/src/main/java/com/microsoft/identity/client/internal/MsalUtils.java

```diff
@@ -234,8 +237,26 @@ public static int getExpiryOrDefault(String expiresIn) {
      * @param url
      * @return
      */
+    @Deprecated
     public static boolean hasCustomTabRedirectActivity(@NonNull final Context context,
                                                        @NonNull final String url) {
+        try {
+            return validateCustomTabRedirectActivity(context, url);
+        } catch (final MsalClientException e) {
+            return false;
```

Comment:
I see that, based on the boolean you return the exception, a bit interesting to not pass on the exact exception?

Reply:
This is the API I deprecated...I don't want to modify the behavior of the existing API as it is Public API...or we could choose to do it...I think this wasn't meant to be called by customers so not sure why it is Public...but now that it was public..I chose not to change the behavior

Reply:
I changed it back to the original code in the deprecated method and opted to have code duplication instead so we can easily remove it in the future. Addressed here: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1357/commits/91fe570fc177b9cc298d2d85147cebff4f98344d